src/sentry/api/endpoints/organization_release_assemble.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+
+import jsonschema
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationReleasesBaseEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.models import Release
+from sentry.tasks.assemble import get_assemble_status, set_assemble_status, \
+    AssembleTask, ChunkFileState
+from sentry.utils import json
+
+
+class OrganizationReleaseAssembleEndpoint(OrganizationReleasesBaseEndpoint):
+    def post(self, request, organization, version):
+        """
+        Handle an artifact bundle and merge it into the release
+        ```````````````````````````````````````````````````````
+
+        :auth: required
+        """
+
+        try:
+            release = Release.objects.get(
+                organization_id=organization.id,
+                version=version,
+            )
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if not self.has_release_permission(request, organization, release):
+            raise ResourceDoesNotExist
+
+        schema = {
+            "type": "object",
+            "properties": {
+                "checksum": {
+                    "type": "string",
+                    "pattern": "^[0-9a-f]{40}$",
+                },
+                "chunks": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "pattern": "^[0-9a-f]{40}$",
+                    }
+                }
+            },
+            "required": ["checksum", "chunks"],
+            "additionalProperties": False,
+        }
+
+        try:
+            data = json.loads(request.body)
+            jsonschema.validate(data, schema)
+        except jsonschema.ValidationError as e:
+            return Response({'error': str(e).splitlines()[0]},
```

Comment:
Could the error contain unicode?

Reply:
No idea, to be honest. I think it usually only contains names from the schema in predefined messages, but we could take a look. 
This way of dealing with validation errors is identical to many other places however, so I assume that it would work correctly. 
Also, the payloads here are generated by `sentry-cli` only, this endpoint is not intended to be used manually.

Reply:
Agreed. This is not worse to how we do it elsewhere :P